### PR TITLE
Blog post on DOT

### PR DESCRIPTION
**Pull request by odersky**
Comments solicited, but not merge yet.

### CODE REVIEW

File: blog/_posts/2016-02-03-essence-of-scala.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+get `String = A = Int`, hence by transitivity `String = Int`! Once you
+are that far, you can of course engineer all sorts of situations where
+a program will typecheck but cause a wrong execution at runtime. In
+other words, type soundness is violated.
+
+Now, the problem is that one cannot always detect these
+inconsistencies, at least not by a local analysis that does not need
+to look at the whole program. What's worse, once you have an
+inconsistent set of definitions you can use these definitions to
+"prove" their own consistency - much like an mathematical theory that
```
Comment (namin) [inline blog/_posts/2016-02-03-essence-of-scala.md]: an -> a

File: blog/_posts/2016-02-03-essence-of-scala.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    x      for (immutable) variables
+    a      for (parameterless) methods
+    A      for types
+
+The full calculus adds to this syntax formal _typing rules_ that assign
+types `T` to terms `t` and formal _evaluation rules_ that describe how a
+program is evaluated. The main difference between the Wadlerfest paper
+and the prior technical report lies in the form of these evaluation
+rules. The technical report uses a "big step" semantics, i.e. it
+defines an abstract of interpreter that explains how terms map to
```
Comment (namin) [inline blog/_posts/2016-02-03-essence-of-scala.md]: "an abstract of interpreter"?
did you mean, "an abstract interpreter"
is "abstract" really needed?

Comment (odersky) [inline blog/_posts/2016-02-03-essence-of-scala.md]: I meant "abstract interpreter". But fine with shortening to "interpreter".

File: blog/_posts/2016-02-03-essence-of-scala.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+studied in depth before. In DOT - as well as in many programming languages -
+you can have conflicting definitions. For instance you might have an abstract
+type declaration in a base class with two conflicting aliases in subclasses:
+
+     trait Base { type A }
+     trait Sub1 extends Base { type A = String }
+     trait Sub2 extends Base { type A = Int }
+
+Now, if you combine `Sub1` and `Sub2` in one class, you get a conflict, since the type `A` is supposed to be equal to both `String` and `Int`. If you do
+not detect the conflict and assume the equalities at face value you
```
Comment (samuelgruetter) [inline blog/_posts/2016-02-03-essence-of-scala.md]: "Now, if you combine Sub1 and Sub2 in one class" is a crucial step and I guess readers will want to know more precisely what you mean here, e.g. some Scala code like `trait Bad extends Sub1 with Sub2`.

**Comment by SethTisue** [comment]
You might consider putting the "Why is This Important?" section first, not last, if you want that section to be widely read. In the current order, a lot of general readers will bail during the calculus section.

File: blog/_posts/2016-02-03-essence-of-scala.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+The following _type soundness_ property was shown in each case with a
+mechanized proof:
+
+> If a term `t` has type `T`, and the evaluation of `t` terminates, then
+  the result of the evaluation will be a value `v` of type `T`.
+
+## Difficulties
+
+Formulating the precise soundness theorem and proving it was unexpectedly hard,
+because it uncovered a problem complex that had not been
```
Comment (smarter) [inline blog/_posts/2016-02-03-essence-of-scala.md]: "a problem complex" ? Do you me "a complex problem" ?

**Comment by odersky** [comment]
@SethTisue It's a concern but it would be difficult to change the flow now. Let's leave it as it is.

**Comment by odersky** [comment]
I believe it's ready to merge now, unless there are further change requests.

**Pull request merged**